port_ocean/core/defaults/initialize.py

```diff
@@ -50,18 +53,52 @@ def deconstruct_blueprints_to_creation_steps(
     )
 
 
-async def _create_resources(
+async def _initialize_required_integration_settings(
     port_client: PortClient,
-    defaults: Defaults,
+    default_mapping: PortAppConfig,
     integration_config: IntegrationConfiguration,
 ) -> None:
-    response = await port_client._get_current_integration()
-    if response.status_code == status.HTTP_404_NOT_FOUND:
-        logger.info("Integration doesn't exist, creating new integration")
-    else:
-        logger.info("Integration already exists, skipping integration creation...")
-        return
+    try:
+        logger.info("Initializing integration at port")
+        integration = await port_client.get_current_integration(
+            should_log=False, should_raise=False
+        )
+        if not integration:
+            integration = await port_client.create_integration(
+                integration_config.integration.type,
+                integration_config.event_listener.to_request(),
+                port_app_config=default_mapping,
+            )
+        elif not integration["config"]:
+            integration = await port_client.patch_integration(
+                integration_config.integration.type,
+                integration_config.event_listener.to_request(),
+                port_app_config=default_mapping,
+            )
+    except httpx.HTTPStatusError as err:
+        logger.error("Failed to apply default mapping")
+        raise err
+
+    logger.info("Checking for diff in integration configuration")
+    changelog_destination = integration_config.event_listener.to_request().get(
+        "changelog_destination"
+    )
+    if (
+        integration["changelogDestination"] != changelog_destination
+        or integration["installationAppType"] != integration_config.integration.type
+        or integration.get("version") != port_client.integration_version
+    ):
+        await port_client.patch_integration(
+            integration_config.integration.type,
+            changelog_destination,
+            default_mapping,
+        )
```

Comment:
why is this? I don't understand why we would like to override the integration config if there is a diff

Reply:
i copied it from `port_ocean/clients/port/mixins/integrations.py`, not sure why is that exactly but i dont wanna change the current behavior regarding this, maybe @yairsimantov20 knows more

Reply:
@omby8888 you can omit the default mapping from there
Just update the version, type and changelog destination